tensorpipe/python/tensorpipe.cc

```diff
@@ -187,9 +187,9 @@ std::shared_ptr<IncomingMessage> prepareToAllocate(
   std::vector<std::shared_ptr<IncomingTensor>> pyTensors;
   pyTensors.reserve(tpMessage.tensors.size());
   for (const auto& tpTensor : tpMessage.tensors) {
-    TP_DCHECK(tpTensor.data == nullptr);
-    pyTensors.push_back(
-        std::make_shared<IncomingTensor>(tpTensor.length, tpTensor.metadata));
+    TP_DCHECK(tpTensor.data.cpu.data == nullptr);
```

Comment:
Maybe even add a check that it is indeed a CPU tensor?

Reply:
For some reason, adding `TP_DCHECK(tpTensor.data.type == tensorpipe::DeviceType::kCpu);` causes:
```
lto1: internal compiler error: in add_symbol_to_partition_1, at lto/lto-partition.c:155
Please submit a full bug report,
with preprocessed source if appropriate.
See <https://bugs.gentoo.org/> for instructions.
lto-wrapper: fatal error: /usr/bin/c++ returned 1 exit status
compilation terminated.
/usr/lib/gcc/x86_64-pc-linux-gnu/8.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: error: lto-wrapper failed
collect2: error: ld returned 1 exit status
```

Reply:
Well that's weird... Do other compilers also fail? Did you report the bug? Adding such a check is not a blocker but it would be nice and I'm also curious about getting to the bottom of this...